spotify-artist-viewer/SAArtistViewController.m

```diff
@@ -0,0 +1,110 @@
+//
+//  SAArtistViewController.m
+//  spotify-artist-viewer
+//
+//  Created by Randall Spence on 6/9/15.
+//  Copyright (c) 2015 Intrepid. All rights reserved.
+//
+
+#import "SAArtistViewController.h"
+#import "SAArtist.h"
+#import "SASearchViewController.h"
+#import <QuartzCore/QuartzCore.h>
+@interface SAArtistViewController () <UIScrollViewDelegate>
+@property (strong, nonatomic) IBOutlet UIScrollView *scrollView;
+
+@end
+
+@implementation SAArtistViewController
+
+static NSString * const SAArtistsSearchUrl = @"http://developer.echonest.com/api/v4/artist/biographies?api_key=KUAKSGNM0ERIJ3ZO8&id=spotify:artist:%@";
+
+-(void)viewWillAppear:(BOOL)animated{
+    [super viewWillAppear:animated];
+    
+    self.artistNameLabel.text = self.artist.name;
+    
+    NSData *imageData = [NSData dataWithContentsOfURL:[NSURL URLWithString:self.artist.imageURL]];
+    
+    self.artistImageView.image = [UIImage imageWithData:imageData];
+    
+    self.artistImageView.layer.cornerRadius = self.artistImageView.frame.size.height / 2;
+    self.artistImageView.layer.masksToBounds = YES;
+    self.artistImageView.layer.borderWidth = 0;
+    
+    [self textViewDidChange:self.artistBioTextView];
+    [self.artistBioTextView setText:self.artist.bio];
+    
+    CGFloat scrollViewHeight = 0.0f;
+    for (UIView* view in self.scrollView.subviews)
+    {
+        scrollViewHeight += view.frame.size.height;
+    }
+    CGRect screenRect = [[UIScreen mainScreen] bounds];
+    CGFloat screenWidth = screenRect.size.width;
+    
+    [self.scrollView setContentSize:(CGSizeMake(screenWidth, scrollViewHeight + 50))];
+}
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    // Do any additional setup after loading the view from its nib.
+    UIBarButtonItem *leftBarButtonItem = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemAdd target:self action:@selector(goBack)];
+    self.navigationItem.leftBarButtonItem = leftBarButtonItem;
+    self.artist.bio = [self retrieveBio];
+    self.artistBioTextView.text = self.artist.bio;
+    
+    self.scrollView.delegate = self;
+    
+}
+
+-(instancetype)initWithArtist:(SAArtist*)artist{
+    self = [super init];
+    if(self){
+        self.artist = artist;
+        
+    }
+    return self;
+}
+
```

Comment:
Reorder the init, viewDidLoad, viewWillAppear methods. List them in order of the view lifecycle.